upload-ami/src/upload_ami/smoke_test.py

```diff
@@ -31,7 +31,7 @@ def smoke_test(image_id: str, run_id: str, cancel: bool) -> None:
             MinCount=1,
             MaxCount=1,
             ClientToken=image_id + run_id if run_id else image_id,
-            InstanceMarketOptions={"MarketType": "spot"},
+            InstanceMarketOptions=instance_market_options,
```

Comment:
I think this doesn't work.  This is a type-error.  `InstanceMarketOptions` takes a `TypedDict[InstanceMarketOptionsType]` not a string.

Reply:
How about a boolean `--use-spot` instead or something?

Reply:
Erm, yeah, you're right, I failed to squash the last change before I pushed.

Wasn't sure whether to go for simple or flexible. Happy to change it to a boolean, though.

Reply:
35 minutes to deal with mypy_boto3 not showing how to import type_defs

Now argumentparser is treating my bool flag as true even when I specify false. :upside_down_face: wow.

Reply:
Okay, I added `--no-spot`. I can't deal with argparser right now to make `--spot` work the way I'd expect it to just work.

If `--no-spot` isn't okay, let me know, I'll try again tomorrow.